.github/workflows/benchmark.yaml

```diff
@@ -0,0 +1,72 @@
+name: Benchmarks
+
+on:
+  pull_request:
+    branches: [ '**' ]
+    types: [opened, synchronize, reopened, labeled]
+
+env:
+  PROMSCALE_LOGGING: false
+  SKIP_PROMQL_BENCHMARKS: true
+  golang-version: 1.18.4
+
```

Comment:
nit:

You can use top level permission attribute to gain write access to pull-request., this will enable forks to run benchmark. For more info https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions

```
permissions:
  pull-requests: write
```

Reply:
I'm not sure that's true. From what I understand, the problem is that the workflow from the fork doesn't have access to the secrets from the source repository. From [here](https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks):

> workflows from forks do not have access to sensitive data such as secrets

Reply:
That is partly correct, but GITHUB_TOKEN is a transient/generated token who's scope can be altered. However if you define your own secret, it won't be available in the context of PR created from a fork.

Reply:
You're right, thank you for correcting me! I found this document which explains how it works quite well: https://docs.github.com/en/actions/security-guides/automatic-token-authentication.

Although that document goes on to say: 

> You can use the permissions key to add and remove read permissions for forked repositories, but typically you can't grant write access. The exception to this behavior is where an admin user has selected the **Send write tokens to workflows from pull requests** option in the GitHub Actions settings. For more information, see "[Managing GitHub Actions settings for a repository](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#enabling-workflows-for-private-repository-forks)."

Reply:
I made a test [PR](https://github.com/timescale/promscale/pull/1517) with this, and it seems to give the [same error](https://github.com/timescale/promscale/runs/7466734734?check_suite_focus=true). Did I mess up something?

Reply:
The quote I made above is relevant here:

> You can use the permissions key to add and remove read permissions for forked repositories, but typically you can't grant write access. The exception to this behavior is where an admin user has selected the **Send write tokens to workflows from pull requests** option in the GitHub Actions settings.

Note that it's not possible to enable `Send write tokens to workflows from pull requests` for public repos (according to this: https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#enabling-workflows-for-forks-of-private-repositories.